### PR TITLE
Plugin Creators now saves user settings

### PR DESCRIPTION
**Pull request by Marr11317**
The function PluginCreator.writeSettings existed, but was never called. It is now called when the window's x button is triggered, even, if the action is canceled.

### CODE REVIEW

**Comment by lasconic** [comment]
Thank you for your pull request. I will need you to sign the CLA at https://musescore.org/en/cla
Could you also squash the commits? gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

File: mscore/pluginCreator.cpp [hunk 1]
```diff
@@ -240,20 +240,22 @@ void PluginCreator::readSettings()
...
+            if (n == QMessageBox::Save){
+                  savePlugin();
+                  }
             else if (n == QMessageBox::Cancel) {
                   ev->ignore();
                   return;
                   }
-            }
+            }
+      PluginCreator.writeSettings();
```
Comment (lasconic) [inline mscore/pluginCreator.cpp]: This is not C++ and you probably didn't try to compile your code !

**Comment by lasconic** [comment]
Hi, I assume your MuseScore username is marr11317 correct ?
Can you squash the commits please ? http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html and force push the result in the same branch. Also, there is a lot of white space changes in your commits, can you make sure  that you only change the lines that need to be changed to fix the problem?  See https://github.com/musescore/MuseScore/pull/3568/files

File: mscore/pluginCreator.cpp [hunk 2]
```diff
@@ -240,20 +240,22 @@ void PluginCreator::readSettings()
...
+      {
       if (state == PCState::DIRTY) {
             QMessageBox::StandardButton n = QMessageBox::warning(this, tr("MuseScore"),
                tr("Plugin \"%1\" has changes.\n"
                "Save before closing?").arg(path),
                QMessageBox::Save | QMessageBox::Discard | QMessageBox::Cancel,
                QMessageBox::Save);
-            if (n == QMessageBox::Save)
-                  savePlugin();
+            if (n == QMessageBox::Save){
```
Comment (Jojo-Schmitz) [inline mscore/pluginCreator.cpp]: Space missing between ) and {
But then again those curly braces are not need at all

Comment (Marr11317) [reply mscore/pluginCreator.cpp]: But from https://musescore.org/en/handbook/musescore-coding-rules
"Exception 2: Use braces also in if-then-else blocks where either the if-code or the else-code covers several lines:

if (address.isEmpty()) {
      --it;
      } 
else {
      qDebug("%s", qPrintable(address));
      ++it;
      }"
I will just do like you tell me

Comment (Marr11317) [reply mscore/pluginCreator.cpp]: okay the spaces didn't copy in previous post, sorry for that

**Comment by Marr11317** [comment]
sorry, I am trying to squash my commits unsuccesfully. 
Sorry for all the complications for such a simple modification.

**Comment by Marr11317** [comment]
I will close this pull request, redo my things better, and give it another try.
Sorry again

**Pull request closed without merging**